src/core/components/ChatHistory.tsx

```diff
@@ -229,13 +235,19 @@ const HistoryItem = memo(
 
     useEffect(() => {
       if (editingTitle) {
+        const conversations = JSON.parse(
+          localStorage.getItem('conversations') ?? '{}',
+        );
+        const currentTitle = conversations[id]?.title ?? title;
+        setNewTitle(currentTitle);
+
```

Comment:
The transition of the title from "New Chat" ➡️ "Summarization" won't trigger any react re-render, so we should double check local storage here to be sure we have the most current value for the title